WooCommerce/Classes/ViewRelated/Coupons/CouponListViewController.swift

```diff
@@ -201,6 +239,7 @@ extension CouponListViewController: UITableViewDelegate {
             self.navigationController?.popViewController(animated: true)
             let notice = Notice(title: Localization.couponDeleted, feedbackType: .success)
             self.noticePresenter.enqueue(notice: notice)
+            self.viewModel.onCouponDeleted()
```

Comment:
I noticed that when we delete the last coupon from the list when don't notify the view model about it so it can change the list state back to `.empty`, leading to a blank view with no information. Since we now have a new empty list view, I decided to fix this here by defining the `onCouponDeleted` function.